packages/app/src/locale/use-lokalize-text.tsx

```diff
@@ -0,0 +1,193 @@
+ *
+ * The live/paths state can also be toggled with shortkey shift+t.
+ */
+export function useLokalizeText(
+  locale: LanguageKey,
+  { enableHotReload }: { enableHotReload?: boolean }
+) {
+  const isMountedRef = useIsMountedRef();
+  const [displayMode, setDisplayMode] = useState<'path' | 'live'>();
+
+  useHotkey(
+    'shift+t',
+    () => setDisplayMode((x) => (x === 'path' ? 'live' : 'path')),
+    {
+      isDisabled: !(enableHotReload && isDefined(displayMode)),
+      disableTextInputs: true,
+    }
+  );
+
+  const [text, setText] = useState(languages[locale]);
+  const textRef = useRef(text);
+  useEffect(() => {
+    textRef.current = text;
+  }, [text]);
+
+  useEffect(() => {
+    if (!enableHotReload || !isDefined(displayMode)) {
+      setText(languages[locale]);
+      return;
+    }
+
+    const setTextDebounced = debounce(setText, 500);
+    const query = `*[_type == 'lokalizeText']`;
+
+    let subscription: Subscription | undefined;
+
+    getClient().then(async (client) => {
+      await client
+        .fetch(query)
+        .then((documents: LokalizeText[]) => {
+          if (!isMountedRef.current) return;
+          const flattenTexts = createFlattenTexts(documents)[locale];
+          setText(() => unflatten(flattenTexts, { object: true }));
+        })
+        .catch((err) => console.error(err));
+
+      if (!isMountedRef.current) return;
+
+      subscription = client
+        .listen(query)
+        .subscribe((update: MutationEvent<LokalizeText>) => {
+          if (!isMountedRef.current) return;
+
+          /**
+           * we currently only handle "updates" to existing (draft) documents.
+           * This means for example that the "discard changes" action is not
+           * handled.
+           */
+          if (!update.result) return;
+
+          const { key, localeText } = parseLocaleTextDocument(update.result);
+
+          /**
+           * We'll mutate text which lives in a reference and update the text
+           * state with a debounced handler. Otherwise the app can become quite
+           * slow when someone is typing in sanity lokalizeText documents.
+           */
+          set(textRef.current, key, localeText[locale]);
+          setTextDebounced(() => JSON.parse(JSON.stringify(textRef.current)));
+        });
+    });
+
+    return () => {
+      subscription?.unsubscribe();
+    };
+  }, [enableHotReload, displayMode, locale, isMountedRef]);
+
+  const paths = useMemo(
+    () =>
+      displayMode === 'path'
+        ? /**
+           * Here we map all locale text object values to their "path"
+           */
+          (unflatten(
+            Object.keys(flatten(text)).reduce((res, key) => {
+              res[key] = key.replace(/\./g, '::');
+              return res;
+            }, {} as Record<string, string>),
+            { object: true }
+          ) as typeof text)
+        : undefined,
+    [displayMode, text]
+  );
+
+  const toggleEl = enableHotReload ? (
```

Comment:
nit; To me this abbreviation is not so common so I wonder if it's worth it.

```suggestion
  const toggleElement = enableHotReload ? (
```